addons/payment_test/controllers/main.py

```diff
@@ -5,19 +5,28 @@
 
 
 class PaymentTestController(http.Controller):
+    _simulation_url = '/payment/test/simulate_payment'
 
-    @http.route('/payment/test/simulate_payment', type='json', auth='public')
-    def test_simulate_payment(self, reference, customer_input):
+    @http.route(_simulation_url, type='json', auth='public')
+    def test_simulate_payment(self, **data):
         """ Simulate the response of a payment request.
 
-        :param str reference: The reference of the transaction
-        :param str customer_input: The payment method details
+        :param dict data: The notification data built with fake objects.
+        See `_process_notification_data`.
         :return: None
         """
-        fake_api_response = {
-            'reference': reference,
-            'cc_summary': customer_input[-4:],
-        }
-        request.env['payment.transaction'].sudo()._handle_notification_data(
-            'test', fake_api_response
+        # Retrieve the tx and acquirer based on the tx reference included in the return url
+        tx_sudo = request.env['payment.transaction'].sudo()._get_tx_from_notification_data(
+            'test', data
         )
+        acquirer_sudo = tx_sudo.acquirer_id
+
+        if acquirer_sudo.capture_manually and data['status'] == 'done':
+            data['status'] = 'authorized'
```

Comment:
I can't do that because when the payment is authorized, the capture button will send a `simulated_state` "done" and the transaction will never be captured because `self.capture_manually` will always stay "True" and the `simulated_state` will be set to `authorized` each time.

If I do something like that:
```
elif state == 'done':
    if self.capture_manually:
        self._set_authorized()
    else:
        self._set_done()
        if self.operation == 'refund':
            self.env.ref('payment.cron_post_process_payment_tx')._trigger()
```

Reply:
I see but the action method could have passed an extra data key to make the distinction. Now it's not a very big deal so let's keep it like this.